petastorm/pytorch.py

```diff
@@ -277,6 +284,28 @@ def __iter__(self):
 
         keys = None
 
+        if self._in_iter:
```

Comment:
nit: maybe it can make the code clearer, if we are more explicit about the fact that this is a three state and not just a boolean, i.e. `if self._in_iter is not None and self._in_iter == True:`?